R/PseudoBulk.R

```diff
@@ -88,8 +89,11 @@ PseudobulkSeurat <- function(seuratObj, groupFields, assays = NULL, additionalFi
 DesignModelMatrix <- function(seuratObj, contrast_columns, sampleIdCol = "cDNA_ID"){
   #Create a dummy sce@colData that unites the contrast columns into a single "group" column
   #combined columns are (by default) separated by an underscore
+  
   colData_intermediate <- seuratObj@meta.data |>
     as.data.frame() |>
+    select(all_of(contrast_columns)) |>
+    mutate(dplyr::across(everything(), \(x) gsub(x,pattern = "_", replacement = "."))) |>
```

Comment:
@bbimber - I wanted to highlight some of the things we discussed that I changed in this last commit. This front loads establishing an underscore as the delimiter downstream by `gsub`-ing them to periods at the start of the "contrast" pipeline. 

I suspect this specific line will degrade eventually as dplyr tends to depreciate verbs regularly.